src/Compilers/Core/Portable/Operations/OperationExtensions.Clone.cs

```diff
@@ -0,0 +1,477 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Semantics
+{
+    public static partial class OperationExtensions
+    {
+        private class Cloner : OperationVisitor<object, IOperation>
```

Comment:
I'm skeptical if cloning is necessary.  The one case we talked aobut yesterday was just ILocalReferenceExpression.  In that case we could just literally make copies of those specific nodes.  I don't know when we'd need generalizing cloning.